### PR TITLE
Hardware and battery package

### PR DESCRIPTION
**Pull request by KmakD**
- Added panther_battery package with nodes copied and renamed from panther_driver repository  for 2 panther versions
- Added panther_power_control package with nodes copied and renamed from panther_driver repository  for 2 panther versions
- updated docs

### CODE REVIEW

File: panther_power_control/package.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>panther_power_control</name>
+  <version>0.0.0</version>
+
+  <description>The panther_power_control package</description>
+  <license>Apache License 2.0</license>
+
+  <author email="dawid.kmak@husarion.com">Dawid Kmak</author>
+  <maintainer email="contact@husarion.com">Husarion</maintainer>
```
Comment (Kotochleb) [inline panther_power_control/package.xml]:
```suggestion
  <maintainer email="support@husarion.com">Husarion</maintainer>
```

File: panther_power_control/package.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0"?>
+<package format="2">
```
Comment (Kotochleb) [inline panther_power_control/package.xml]:
```suggestion
<package format="3">
```

Let's try using format 3 everywhere we can

File: panther_power_control/package.xml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  <name>panther_power_control</name>
+  <version>0.0.0</version>
```
Comment (Kotochleb) [inline panther_power_control/package.xml]:
```suggestion
  <version>0.0.1</version>
```

Maybe some starting version?

File: panther_power_control/package.xml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  <description>The panther_power_control package</description>
```
Comment (Kotochleb) [inline panther_power_control/package.xml]:
```suggestion
  <description>Nodes responsible for controling power section of Husarion Panther robot</description>
```

File: panther_power_control/package.xml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  <maintainer email="contact@husarion.com">Husarion</maintainer>
+
+  <url type="website">https://husarion.com/</url>
+  <url type="repository">https://github.com/husarion/panther_ros</url>
+  <url type="bugtracker">https://github.com/husarion/panther_ros/issues</url>
+
+  <buildtool_depend>catkin</buildtool_depend>
+  <build_depend>rospy</build_depend>
+  <build_export_depend>rospy</build_export_depend>
+  <exec_depend>rospy</exec_depend>
```
Comment (Kotochleb) [inline panther_power_control/package.xml:19]: Can't it just be dependent?

File: panther_power_control/package.xml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+  <depend>std_msgs</depend>
+  <depend>std_srvs</depend>
+  <depend>python-rpi.gpio-pip</depend>
+  <depend>python3-gpiozero</depend>
```
Comment (Kotochleb) [inline panther_power_control/package.xml]: Imports are not in alphabetical order

File: panther_power_control/launch/power_control.launch [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<launch>
+
+</launch>
```
Comment (Kotochleb) [inline panther_power_control/launch/power_control.launch:3]: Why is it empty?

File: panther_power_control/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,16 @@
+cmake_minimum_required(VERSION 3.0.2)
+project(panther_power_control)
+
+find_package(catkin REQUIRED COMPONENTS
+  rospy std_msgs std_srvs
+)
```
Comment (Kotochleb) [inline panther_power_control/CMakeLists.txt]:
```suggestion
find_package(catkin REQUIRED COMPONENTS
  rospy
  std_msgs
  std_srvs
)
```

File: panther_battery/package.xml [hunk 9]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>panther_battery</name>
+  <version>0.0.0</version>
+
+  <description>The panther_battery package</description>
+  <license>Apache License 2.0</license>
+
+  <author email="dawid.kmak@husarion.com">Dawid Kmak</author>
+  <maintainer email="contact@husarion.com">Husarion</maintainer>
```
Comment (Kotochleb) [inline panther_battery/package.xml]:
```suggestion
  <maintainer email="support@husarion.com">Husarion</maintainer>
```

File: panther_battery/package.xml [hunk 10]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>panther_battery</name>
+  <version>0.0.0</version>
+
+  <description>The panther_battery package</description>
```
Comment (Kotochleb) [inline panther_battery/package.xml]:
```suggestion
  <description>Nodes monitoring the battery state of Husarion Panhter robot</description>
```

File: panther_battery/package.xml [hunk 11]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0"?>
+<package format="2">
```
Comment (Kotochleb) [inline panther_battery/package.xml]:
```suggestion
<package format="3">
```

File: panther_battery/package.xml [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+  <name>panther_battery</name>
+  <version>0.0.0</version>
```
Comment (Kotochleb) [inline panther_battery/package.xml]: Some non zero version here would be nice

File: panther_battery/package.xml [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+  <maintainer email="contact@husarion.com">Husarion</maintainer>
+
+  <url type="website">https://husarion.com/</url>
+  <url type="repository">https://github.com/husarion/panther_ros</url>
+  <url type="bugtracker">https://github.com/husarion/panther_ros/issues</url>
+
+  <buildtool_depend>catkin</buildtool_depend>
+  <build_depend>rospy</build_depend>
+  <build_export_depend>rospy</build_export_depend>
+  <exec_depend>rospy</exec_depend>
```
Comment (Kotochleb) [inline panther_battery/package.xml:19]: Maybe all as a `dependent` tag??

File: panther_battery/launch/battery.launch [hunk 14]
```diff
@@ -0,0 +1,3 @@
+<launch>
+
+</launch>
```
Comment (Kotochleb) [inline panther_battery/launch/battery.launch:3]: Another empty launch

File: panther_battery/CMakeLists.txt [hunk 15]
```diff
@@ -0,0 +1,16 @@
+cmake_minimum_required(VERSION 3.0.2)
+project(panther_battery)
+
+find_package(catkin REQUIRED COMPONENTS
+  rospy sensor_msgs panther_msgs
+)
```
Comment (Kotochleb) [inline panther_battery/CMakeLists.txt]:
```suggestion
find_package(catkin REQUIRED COMPONENTS
  rospy
  panther_msgs
  sensor_msgs
)
```

File: README.md [hunk 16]
```diff
@@ -9,4 +9,72 @@ Metapackeage that contains dependencies to other repositories.
...
+  - `/panther_hardware/e_stop_reset` [*std_srvs/Trigger*]: this service allows to reset emergency stop
+  - `/panther_hardware/e_stop_trigger` [*std_srvs/Trigger*]: this service allows to trigger emergency stop
+
+- **`relays_node.py`**
+
+  This node is responsible for power management using relays available in Panther versions 1.05 and 1.06
+
+  **Topics published**
+
+  - `/panther_hardware/motor_on` [*std_msgs/Bool*]: informs if motor controllers are on
```
Comment (Kotochleb) [inline README.md]: Update formatting to what we discussed

**Review by Kotochleb — CHANGES_REQUESTED** [review]

File: panther_power_control/package.xml (hunk 5 above)
Comment (KmakD) [reply panther_power_control/package.xml:19]: those are automatically generated with `catkin_create_pkg [pkg_name] rospy`

File: panther_power_control/launch/power_control.launch (hunk 7 above)
Comment (KmakD) [reply panther_power_control/launch/power_control.launch:3]: It is just for package structure. Launch is to be written as part of a different task

File: panther_power_control/package.xml (hunk 1 above)
Comment (KmakD) [reply panther_power_control/package.xml]: `panther` and `panther_brinup` packages also have contact in package.xml will change that too.

File: panther_power_control/package.xml (hunk 2 above)
Comment (KmakD) [reply panther_power_control/package.xml]: Isn't this more ROS2 focused? Acording to ROS1 documentation format 2 is recomended http://wiki.ros.org/catkin/package.xml

Comment (Kotochleb) [reply panther_power_control/package.xml]: Format 3 is newer and backward compatible with format 2. This dock might be also outdated, as many things in ROS...

File: panther_power_control/package.xml (hunk 5 above)
Comment (Kotochleb) [reply panther_power_control/package.xml:19]: Ok. Then we can leave it as it is

File: README.md [hunk 17]
```diff
@@ -1,12 +1,89 @@
...
+
+### `relays_node.py`
+
+This node is responsible for power management using relays available in Panther versions 1.05 and 1.06
+
+#### Topics published
 
-Package that contains default robot config and launch, which starts all base functionalities and parameters.
\ No newline at end of file
+- `/panther_hardware/motor_on` [*std_msgs/Bool*]: informs if motor controllers are on
```
Comment (Kotochleb) [inline README.md]: My suggestions are shown in here: https://github.com/husarion/panther_ros/pull/5

**Review by Kotochleb — CHANGES_REQUESTED** [review]

File: README.md [hunk 18]
```diff
@@ -1,12 +1,84 @@
-# panther_ros
-ROS package for Panther autonomous mobile robot
+# ROS packages
```
Comment (Kotochleb) [inline README.md]: One last thing. Let's remove this line
```suggestion
```

**Review by Kotochleb — CHANGES_REQUESTED** [review]

**Review by Kotochleb — APPROVED** [review]

**Pull request merged**